pygmt/src/meca.py

```diff
@@ -213,6 +213,7 @@ def meca(
     # pylint: disable=too-many-nested-blocks
     # pylint: disable=too-many-branches
     # pylint: disable=too-many-statements
+    # pylint: disable=consider-using-dict-items
```

Comment:
Might be better to take this opportunity to refactor things a little bit. Below is a suggestion, I tried looping over `param_conventions.items()` but encountered a different pylint error (undefined-loop-variable), so did `list(param_conventions)` instead. Also refactored the chunk to use Python's [for-else](https://book.pythontips.com/en/latest/for_-_else.html#else-clause).

```diff
diff --git a/pygmt/src/meca.py b/pygmt/src/meca.py
index 61af2457..f0fed359 100644
--- a/pygmt/src/meca.py
+++ b/pygmt/src/meca.py
@@ -213,7 +213,6 @@ def meca(
     # pylint: disable=too-many-nested-blocks
     # pylint: disable=too-many-branches
     # pylint: disable=too-many-statements
-    # pylint: disable=consider-using-dict-items
 
     def set_pointer(data_pointers, spec):
         """
@@ -295,16 +294,14 @@ def meca(
                 spec_conv = spec
 
         # set convention and focal parameters based on spec convention
-        convention_assigned = False
-        for conv in param_conventions:
+        for conv in list(param_conventions):
             if set(spec_conv.keys()) == set(param_conventions[conv]):
                 convention = conv.lower()
                 foc_params = param_conventions[conv]
-                convention_assigned = True
                 break
-        if not convention_assigned:
+        else:  # if there is no convention assigned
             raise GMTError(
-                "Parameters in spec dictionary do not match known " "conventions."
+                "Parameters in spec dictionary do not match known conventions."
             )
 
         # create a dict type pointer for easier to read code
```

Reply:
Yes, this is a good idea. The last couple commits apply your suggestions.